src/features/projectsV2/ProjectsMap/ExploreLayers.tsx

```diff
@@ -0,0 +1,59 @@
+import type { ReactElement } from 'react';
+import type { MapOptions } from '../ProjectsMapContext';
+import type { MapLayerOptionsType } from '../../../utils/mapsV2/mapSettings.config';
+
+import { Layer, Source } from 'react-map-gl-v7/maplibre';
+import { useProjectsMap } from '../ProjectsMapContext';
+
+const getSelectedLayerKeys = (
+  mapOptions: MapOptions
+): MapLayerOptionsType[] => {
+  const selectedLayers: MapLayerOptionsType[] = [];
+  Object.entries(mapOptions).forEach(([key, value]) => {
+    if (key !== 'projects' && value === true) {
+      selectedLayers.push(key as MapLayerOptionsType);
+    }
+  });
+  return selectedLayers;
+};
+
+export default function ExploreLayers(): ReactElement | null {
+  const { exploreLayersData, mapOptions } = useProjectsMap();
+  if (!exploreLayersData) return null;
+
+  const selectedLayers = getSelectedLayerKeys(mapOptions);
+  if (selectedLayers.length === 0) return null;
+
+  return (
+    <>
+      {selectedLayers.map((layerKey) => {
+        const layerData = exploreLayersData[layerKey];
+
+        if (!layerData || !layerData.tileUrl) {
+          return null;
+        }
```

Comment:
Avoid returning `null` inside .map() as it still adds an` empty entry to the Virtual DOM`, which can lead to unnecessary re-renders. Instead, use .filter() before .map() to remove unwanted items upfront. This improves performance and keeps the code cleaner.